src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -97,7 +99,7 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             // No AT in the cache or AT needs to be refreshed
             try
             {
-                if (cachedAccessToken == null)
+                if (!isCachedAccessTokenValid)

```

Comment:
This won't exactly work. RefreshRt method would try to use the cached RT to get a token and AAD will throw. So need to make sure to just call get new AT method right away.